src/drivers/gpio/gpio.h

```diff
@@ -49,6 +49,10 @@ typedef volatile unsigned long gpio_mask_t;
 #define GPIO_MODE_OUT_ALTERNATE     (1 << 13)
 #endif
 
+/* Alternate function number. Bits: 17 ... 20 */
+#define GPIO_GET_ALTERNATE(mode) (((mode) & 0x1e0000) >> 17)
```

Comment:
This is common part or STM32 depends

Reply:
It doesn't depends on stm32. It's a general alternative function number. Each driver could implement this I think

Reply:
So we set GPIO in alternative mode and set also alternative mode.